dialect/sql/schema/atlas.go

```diff
@@ -979,7 +979,9 @@ func (a *Atlas) aIndexes(et *Table, at *schema.Table) error {
 		}
 		pk = append(pk, c2)
 	}
-	at.SetPrimaryKey(schema.NewPrimaryKey(pk...))
+	if len(pk) > 0 {
+		at.SetPrimaryKey(schema.NewPrimaryKey(pk...))
+	}
```

Comment:
Using DiffHooks, the primary keys can be cleaned before generating the changes. Can you try this also? 

```go
schema.WithDiffHook(func(next schema.Differ) schema.Differ {
	return schema.DiffFunc(func(current, desired *atlas.Schema) ([]atlas.Change, error) {
		if t, ok := current.Table(name); ok {
			t.PrimaryKey = nil
		}
		if t, ok := desired.Table(name); ok {
			t.PrimaryKey = nil
		}
		next.Diff(current, desired)
	})
})
```

I don't mind adding this change, but I prefer having an explicit comment explaining that it might be cleared by the user using a `CreateHook` (the old API).